# Mitogenomics_assembly

A bash script has been developed to streamline mitogenomics data analysis using NOVOPlasty with data from NextSeq 2000 sequencing. Here's a quick rundown of what the new script is accomplishing:

𝐅𝐚𝐬𝐭𝐐𝐂: Runs FastQC to visualize data quality reports.
𝐐𝐮𝐚𝐥𝐢𝐭𝐲 𝐂𝐨𝐧𝐭𝐫𝐨𝐥: Uses Trimmomatic to discard reads that fall below quality thresholds and trim low-quality bases.
𝐔𝐬𝐞𝐫 𝐈𝐧𝐩𝐮𝐭: Collects project-specific parameters such as project name, directories, read files, and range of k-mer values to run.
𝐃𝐢𝐫𝐞𝐜𝐭𝐨𝐫𝐲 𝐒𝐞𝐭𝐮𝐩: Creates necessary directories for reference and assembly.
𝐑𝐞𝐟𝐞𝐫𝐞𝐧𝐜𝐞 𝐃𝐨𝐰𝐧𝐥𝐨𝐚𝐝: Fetches reference genome data from BOLD Systems.
𝐄𝐱𝐞𝐜𝐮𝐭𝐢𝐨𝐧: Generates a dynamic configuration file for NOVOPlasty and runs it iteratively across a range of k-mer values.
𝐂𝐢𝐫𝐜𝐮𝐥𝐚𝐫𝐢𝐳𝐚𝐭𝐢𝐨𝐧 𝐂𝐡𝐞𝐜𝐤: Identifies and halts if a circularized genome is assembled.

# Parameters of script "1_trimmomatic":

Trimmed Directory Name: Directory where the trimmed output files will be stored.
Path to the Trimmomatic Jar File: Full path to the Trimmomatic .jar file needed to run the trimming process.
Input R1 File Name: Filename of the input forward read (R1) file from paired-end sequencing.
Input R2 File Name: Filename of the input reverse read (R2) file from paired-end sequencing.
Minimum Length (MINLEN): Minimum length of reads to be kept after trimming; shorter reads are discarded.
Head Crop Length (HEADCROP): Number of bases to crop from the start of each read to remove low-quality bases.
Base Output Name: Base name for the output files generated by Trimmomatic.

# Parameters of script "2_novoplasty":

Project Name: The name of the project to organize and identify different runs.
NOVOPlasty Directory: The directory where NOVOPlasty is installed.
Reference ID: The reference genome ID used for downloading the reference sequence.
Forward Reads File: The file containing the forward reads from the sequencing data.
Reverse Reads File: The file containing the reverse reads from the sequencing data.
Read Length: The length of the reads in the sequencing data.
Insert Size: The size of the insert in the sequencing data.
Start Kmer Value: The starting value of the kmer range for the assembly process.
End Kmer Value: The ending value of the kmer range for the assembly process.
Kmer Step: The step value for iterating through the kmer range.

# Workflow:

https://github.com/fabricioA14/Mitogenomics_assembly/assets/73892283/872e49fe-5d36-4033-a7c9-5deccb53c292






